### DISCUSSION TITLE
discord.errors.ConnectionClosed: Shard ID None WebSocket closed with 1000

### DISCUSSION THREAD

**Discussion by citrus0610** [Rapptz/discord.py#9726] [General]
I run my discord.py bot on my win11 desktop computer 24/7.
However, the following error occurred once in a while.

Attempting a reconnect in 0.30s
Traceback (most recent call last):
File "venv\lib\site-packages\discord\client.py", line 659, in connect
await self.ws.poll_event()
File "venv\lib\site-packages\discord\gateway.py", line 646, in poll_event
raise ConnectionClosed(self.socket, shard_id=self.shard_id, code=code) from None
discord.errors.ConnectionClosed: Shard ID None WebSocket closed with 1000

This error does not affect the program and the program is still running normally, but I want to ignore or catch this error message.
Does anyone know how to ignore or catch the above error?
Thanks~

**Comment by No767** [comment]
@YungPingXu This is normal. See https://github.com/Rapptz/discord.py/discussions/9722#discussioncomment-8400265

**Reply by citrus0610** [reply]
I know this is normal, but I hope my program can ignore the error message in the log.
Unfortunately, I have no idea how to catch the exception for error handling.
Do you have any suggestions?

**Reply by No767** [reply]
You don't need to handle the error. I would recommend you explicitly **not** handle the error as this may warn of further problems. In addition, I'd also advise you to consult this amongst the community for advice. You can find an invite link for the official discord server [here](https://discord.gg/r3sSKJJ)

**Comment by Rapptz** [comment]
If you want to filter specific logs from showing up you need to use a `logging.Filter` on the logger object. If you're using the regular `Client.run` it should be as simple as doing this:

```py
import logging

class GatewayEventFilter(logging.Filter):
    def __init__(self) -> None:
        super().__init__('discord.gateway')

    def filter(self, record: logging.LogRecord) -> bool:
        if record.exc_info is not None and isinstance(record.exc_info[1], discord.ConnectionClosed):
            return False
        return True

logging.getLogger('discord.gateway').addFilter(GatewayEventFilter())

# client related stuff goes here
```
